### PR TITLE
Support for msbuild/build event

### PR DESCRIPTION
**Pull request by rokonec**
Support for telemetry emitted by https://github.com/dotnet/msbuild/pull/7778

This telemetry is supposed to be sent with every build, including base measurements and properties usable to understand usage of MSBuild.

This PR also fixes bug causing not sending telemetry from msbuild - introduced during changes of SDK calling msbuild.dll inproc.

NOTE: Although it is related to MSBuild server it could be merged into SDK before MSBuild server is released.

### CODE REVIEW

**Review by baronfel — APPROVED** [review]
Seems pretty straightforward - glad that it was easy to plumb through here.

File: src/Cli/dotnet/commands/dotnet-msbuild/MSBuildLogger.cs [hunk 1]
```diff
@@ -107,15 +111,47 @@ internal static void FormatAndSend(ITelemetry telemetry, TelemetryEventArgs args
 
                 telemetry.TrackEvent(newEventName, maskedProperties, measurements: null);
             }
+            else if (args.EventName == BuildTelemetryEventName)
+            {
+                var newEventName = $"msbuild/{BuildTelemetryEventName}";
+                Dictionary<string, string> properties = new Dictionary<string, string>(args.Properties);
+                Dictionary<string, double> measurements = new Dictionary<string, double>();
+
+                string[] toBeHashed = new[] { "ProjectPath" };
```
Comment (baronfel) [inline src/Cli/dotnet/commands/dotnet-msbuild/MSBuildLogger.cs:120]: the Target property should be hashed as well - users can and do define custom MSBuild targets that could reveal PII

Comment (rokonec) [inline src/Cli/dotnet/commands/dotnet-msbuild/MSBuildLogger.cs:120]:
```suggestion
                string[] toBeHashed = new[] { "ProjectPath", "BuildTarget" };
```

**Comment by baronfel** [comment]
Anything blocking us from merging this @rokonec?

**Review by marcpopMSFT — APPROVED** [review]

**Pull request merged**